lib/matplotlib/tests/test_legend.py

```diff
@@ -1438,15 +1439,90 @@ def test_legend_text():
 
 
 def test_boxplot_labels():
-    # Test that boxplot(..., labels=) sets the tick labels but not legend entries
-    # This is not consistent with other plot types but is the current behavior.
-    fig, ax = plt.subplots()
+    # Test that legend entries are generated from
+    # boxplots and tick labels are now set using `tick_labels`.
+    # `label` is the new parameter for setting legend labels.
     np.random.seed(19680801)
-    data = np.random.random((10, 3))
-    bp = ax.boxplot(data, labels=['A', 'B', 'C'])
-    # Check that labels set the tick labels ...
-    assert [l.get_text() for l in ax.get_xticklabels()] == ['A', 'B', 'C']
-    # ... but not legend entries
-    handles, labels = ax.get_legend_handles_labels()
-    assert len(handles) == 0
-    assert len(labels) == 0
+    data = np.random.lognormal(size=(68, 4), mean=1.5, sigma=0.2)
+    labels = list('ABCD')
+
+    legend_labels = ['peaches', 'oranges', 'tomatoes', 'plums']
+    fig, axs = plt.subplots(nrows=2,
+                            ncols=3,
+                            figsize=(10, 7),
+                            sharey=True)
+
+    # Should get deprecation warning for `labels`
+    with pytest.warns(mpl.MatplotlibDeprecationWarning,
+                      match='has been renamed \'tick_labels\''):
+        first = axs[0, 0].boxplot(data,
+                                  labels=labels,  # `tick_labels`
+                                  showmeans=True,
+                                  label=legend_labels)
+        axs[0, 0].set_title('Tick and legend labels')
+        axs[0, 0].legend()
+        assert [l.get_text() for l in axs[0, 0].get_xticklabels()] == labels
+        for i, v in enumerate(first['boxes']):
+            assert v.get_label() == legend_labels[i]
+
+    # This plot will not generate a legend.
+    with pytest.warns(UserWarning, match='No artists with labels found'):
+        warnings.warn('No artists with labels found.', UserWarning)
+        second = axs[0, 1].boxplot(data,
+                                   tick_labels=labels,
+                                   showmeans=True,
+                                   patch_artist=True)
+        axs[0, 1].set_title('Tick labels, no legend')
+        axs[0, 1].legend()
```

Comment:
It might be good for this test to [get hold of the legend handles and labels](https://matplotlib.org/devdocs/api/_as_gen/matplotlib.axes.Axes.get_legend_handles_labels.html#matplotlib-axes-axes-get-legend-handles-labels) and assert that these lists are empty.